### DISCUSSION TITLE
Testes Cross-Devices

### DISCUSSION THREAD

**Discussion by vitorbritto** [frontendbr/forum#98] [Perguntas e Respostas]
Boa tarde pessoal. Tudo bem?

Quais ferramentas vocês estão utilizando para testes _Cross-devices_, que replicam fielmente (ou **bem** próximo) a renderização em um _device_ físico?

Testei diversas ferramentas nos últimos 30 dias e confesso que tive uma sucessão de surpresas desagradáveis no resultado final do CSS apresentado no _device_ físico.

Agradeço desde já.

**Comment by bernardodiasc** [comment]
O que voce achou do browserstack @vitorbritto? Chegou a usar esse? Na empresa que trabalho compraram a licença na esperança de resolver esse problema, já que nem emulador de android nem the ios resultavam iguais aos aparelhos reais (bugs específicos em aparelhos específicos com versoes especificas dos sistemas, nao foram casos gerais).

**Comment by vitorbritto** [comment]
@bernardodiasc, enfrentei diversos problemas nos simuladores e emuladores. 

O simulador do iOS no Mac tem suas vantagens. Porém, quanto ao alinhamento na vertical pecou por diversas vezes. Quando tudo parecia bem alinhado no simulador, no device real estava tudo zicado.

Para Android, eu encontrei o Android Player da Xamarim. E cara, posso dizer que foi a salvação por aqui. 

O screenfly não atendeu em nenhum requisito, seja para tablets ou smartphones. O fato é que a altura da janela no [Screenfly](http://quirktools.com/screenfly) leva em consideração as barras do topo e do rodapé. Pelo menos foi o que pude constatar tirando a diferença em pixels para chegar o mais próximo possível da altura do dispositivo físico. Ainda com o Screenfly, as resoluções de TV Screen tiveram diferenças consideráveis também. Quando no Screenfly, o elemento estava alinhado no centro na vertical, na TV de 1080p (1920x1080) estava uns 100px para cima. Testei ainda no Chrome e Firefox, ambos falharam brutalmente. 

Eu criei uma conta no BrowserStack, mas como consegui fazer os testes nos dispositivos reais logo depois, acabei não proseguindo com a compra da licença. Qual foi sua experiência nele? Funciona realmente?

**Comment by bernardodiasc** [comment]
As vezes que testei o BrowserStack funcionou legal, mas para os casos mais complicados ainda não testei. O caso que levou a compra da licença no final das contas devia ser um caso de cache de um dos product owners, então o bug "misteriosamente" sumiu e encerraram o caso. Havendo mais experiências volto aqui pra comentar. Vlw pelo feedback!

**Comment by felipefialho** [comment]
Deixa eu invocar o @neysimoes, que está trabalhando bem focado em mobile.

**Comment by heliomsolivas** [comment]
Ressuscitando a issue, eu uso o browserstack onde eu trabalho (pago) e é muito bom, simulando quase que fiel os devices, a única coisa ruim é o delay na hora de inspecionar os elementos, mas acho que é querer demais, se alguém tiver alguma dica parecida com o browserstack fico feliz em testar.

**Comment by didiraja** [comment]
BrowserStack é bom mas tem um delay enorme,  o melhor disparado que já testei é o [CrossBrowser](https://crossbrowsertesting.com), delay baixo, opção de print e um variedade gigante de de dispositivos, sistemas e resoluções.

Uso esse na agência.

**Comment by felipefialho** [comment]
@didiraja Vou dar uma olhada hein...
